test/Decimal/CPP/DecimalTests.cpp

```diff
@@ -0,0 +1,1563 @@
+            const Microsoft::Windows::Foundation::decimal object(data);
+            const auto to{ object.to_uint64() };
+            VERIFY_ARE_EQUAL(data, to);
+
+            Microsoft::Windows::Foundation::decimal object2;
+            object2 = data;
+            const auto to2{ object.to_uint64() };
+            VERIFY_ARE_EQUAL(data, to2);
+        }
+
+        TEST_METHOD(ctor_to_assign_float)
+        {
+            const float data{ -1.25 };
+            const Microsoft::Windows::Foundation::decimal object(data);
+            const auto to{ object.to_float() };
+            VERIFY_ARE_EQUAL(data, to);
+
+            Microsoft::Windows::Foundation::decimal object2;
+            object2 = data;
+            const auto to2{ object.to_float() };
+            VERIFY_ARE_EQUAL(data, to2);
+        }
+
+        TEST_METHOD(ctor_to_assign_double)
+        {
+            const double data{ -1.25 };
+            const Microsoft::Windows::Foundation::decimal object(data);
+            const auto to{ object.to_double() };
+            VERIFY_ARE_EQUAL(data, to);
+
+            Microsoft::Windows::Foundation::decimal object2;
+            object2 = data;
+            const auto to2{ object.to_double() };
+            VERIFY_ARE_EQUAL(data, to2);
+        }
+
+        TEST_METHOD(ctor_to_assign_long)
+        {
+            const long data{ -1234567890 };
+            const Microsoft::Windows::Foundation::decimal object(data);
+            const auto to{ object.to_long() };
+            VERIFY_ARE_EQUAL(data, to);
+
+            Microsoft::Windows::Foundation::decimal object2;
+            object2 = data;
+            const auto to2{ object.to_long() };
+            VERIFY_ARE_EQUAL(data, to2);
+        }
+
+        TEST_METHOD(ctor_to_assign_ulong)
+        {
+            const unsigned long data{ 1234567890 };
+            const Microsoft::Windows::Foundation::decimal object(data);
+            const auto to{ object.to_ulong() };
+            VERIFY_ARE_EQUAL(data, to);
+
+            Microsoft::Windows::Foundation::decimal object2;
+            object2 = data;
+            const auto to2{ object.to_ulong() };
+            VERIFY_ARE_EQUAL(data, to2);
+        }
+
+        TEST_METHOD(ctor_to_assign_pcwstr)
+        {
+            PCWSTR data{ L"-12.345" };
+            const Microsoft::Windows::Foundation::decimal object(data);
+            const auto to{ object.to_string() };
+            VERIFY_ARE_EQUAL(0, wcscmp(data, to.c_str()));
+
+            Microsoft::Windows::Foundation::decimal object2;
+            object2 = data;
+            const auto to2{ object.to_string() };
+            VERIFY_ARE_EQUAL(0, wcscmp(data, to.c_str()), std::format(L"'{}' == '{}'", data, to.c_str()).c_str());
+        }
+
+        TEST_METHOD(ctor_to_assign_pcwstr_lcid)
+        {
+            PCWSTR data{ L"-12.345" };
+            const Microsoft::Windows::Foundation::decimal object(data, GetSystemDefaultLCID());
```

Comment:
Using `GetSystemDefaultLCID()` or `LOCALE_SYSTEM_DEFAULT` (or other non-specific values) could cause issues for anyone running the test on a machine with different number formats (such as '.' as the grouping symbol [for thousands/millions/etc.]).

Reply:
Hmmm. Good point. The pipeline had no difficulty despite the myriad of environs we run on so yay but I've made a note to revise the tests in the next PR